Build 896 for 7.17 with status FAILURE - FAIL: TestRetryWithBackoff

## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F7.17/detail/7.17/896//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F7.17/detail/7.17/896//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F7.17/detail/7.17/896//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F7.17/detail/7.17/896//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_null.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2023-11-09T16:28:25.774Z&rangeTo=2023-11-09T16:48:25.774Z&transactionName=BUILD+Beats%2Fbeats%2F7.17&transactionType=job&latencyAggregationType=avg&traceId=314faab0a15bf37585494655d469c816&transactionId=4c8ab323ffe71440)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-11-09T16:38:25.774+0000


* Duration: 77 min 12 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 1  |
| Passed       | 27206  |
| Skipped      | 1930 |
| Total        | 29137   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![1](https://img.shields.io/badge/1%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F7.17/detail/7.17/896//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `Build&Test / x-pack/elastic-agent-unitTest / TestRetryWithBackoff – github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/retry`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestRetryWithBackoff
    retrystrategy_test.go:116: [not-fatal-by-iface]: expecting execution time between 350000000-5000000000 ns, got: 313.869676ms
--- FAIL: TestRetryWithBackoff (0.70s)
 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![4](https://img.shields.io/badge/4%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F7.17/detail/7.17/896//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `x-pack/elastic-agent-unitTest - mage build unitTest`
<ul>
<li>Took 2 min 10 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/7.17/runs/896/steps/12981/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `x-pack/elastic-agent-unitTest - mage build unitTest`
<ul>
<li>Took 1 min 46 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/7.17/runs/896/steps/14288/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `x-pack/elastic-agent-unitTest - mage build unitTest`
<ul>
<li>Took 1 min 47 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/7.17/runs/896/steps/14887/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/7.17/runs/896/steps/15573/log/?start=0">here</a></li>
<li>Description: <code>Error "hudson.AbortException: script returned exit code 1"</code></l1>
</ul>
  
  </p>
  </details>

--- a/x-pack/elastic-agent/pkg/core/retry/retrystrategy.go
+++ b/x-pack/elastic-agent/pkg/core/retry/retrystrategy.go
@@ -7,29 +7,7 @@ package retry
 import (
 	"context"
 	"time"
-
-	"github.com/elastic/beats/v7/libbeat/common/backoff"
 )
-
-// DoWithBackoff ignores retry config of delays and lets backoff decide how much time it needs.
-func DoWithBackoff(config *Config, b backoff.Backoff, fn func() error, fatalErrors ...error) error {
-	retryCount := getRetryCount(config)
-	var err error
-
-	for retryNo := 0; retryNo <= retryCount; retryNo++ {
-		err = fn()
-		if err == nil || isFatal(err, fatalErrors...) {
-			b.Reset()
-			return err
-		}
-
-		if retryNo < retryCount {
-			b.Wait()
-		}
-	}
-
-	return err
-}
 
 // Do runs provided function in a manner specified in retry configuration
 func Do(ctx context.Context, config *Config, fn func(ctx context.Context) error, fatalErrors ...error) error {

--- a/x-pack/elastic-agent/pkg/core/retry/retrystrategy_test.go
+++ b/x-pack/elastic-agent/pkg/core/retry/retrystrategy_test.go
@@ -9,8 +9,6 @@ import (
 	"errors"
 	"testing"
 	"time"
-
-	"github.com/elastic/beats/v7/libbeat/common/backoff"
 )
 
 func TestRetry(t *testing.T) {
@@ -72,60 +70,6 @@ func TestRetry(t *testing.T) {
 	}
 }
 
-func TestRetryWithBackoff(t *testing.T) {
-	type testCase struct {
-		Fn                 func() error
-		ExpectedExecutions int
-		IsErrExpected      bool
-		Enabled            bool
-	}
-
-	errFatal := errors.New("fatal")
-	executions := 0
-
-	testCases := map[string]testCase{
-		"not-failing":        testCase{Fn: func() error { executions++; return nil }, ExpectedExecutions: 1, Enabled: true},
-		"failing":            testCase{Fn: func() error { executions++; return errors.New("fail") }, ExpectedExecutions: 4, IsErrExpected: true, Enabled: true},
-		"fatal-by-enum":      testCase{Fn: func() error { executions++; return errFatal }, ExpectedExecutions: 1, IsErrExpected: true, Enabled: true},
-		"fatal-by-iface":     testCase{Fn: func() error { executions++; return ErrFatal{} }, ExpectedExecutions: 1, IsErrExpected: true, Enabled: true},
-		"not-fatal-by-iface": testCase{Fn: func() error { executions++; return ErrNotFatal{} }, ExpectedExecutions: 4, IsErrExpected: true, Enabled: true},
-	}
-
-	config := &Config{
-		RetriesCount: 3,
-		Delay:        5000,
-	}
-	maxDelay := time.Duration(config.Delay) * time.Millisecond
-
-	done := make(chan struct{})
-	maxWaitTime := 200 * time.Millisecond
-	minWaitTime := 50 * time.Millisecond
-	backoff := backoff.NewEqualJitterBackoff(done, minWaitTime, maxWaitTime)
-
-	for n, tc := range testCases {
-		testFn := tc.Fn
-		executions = 0
-		config.Enabled = tc.Enabled
-
-		startTime := time.Now()
-		err := DoWithBackoff(config, backoff, testFn, errFatal)
-
-		executionTime := time.Since(startTime)
-		minExecTime := getBackoffMinTime(minWaitTime, maxWaitTime, tc.ExpectedExecutions)
-		if tc.ExpectedExecutions > 1 && (executionTime < minExecTime || executionTime > maxDelay) {
-			t.Fatalf("[%s]: expecting execution time between %d-%d ns, got: %v", n, minExecTime, maxDelay, executionTime)
-		}
-
-		if (err == nil) == tc.IsErrExpected {
-			t.Fatalf("[%s]: expecting error, got: %v", n, err)
-		}
-
-		if executions != tc.ExpectedExecutions {
-			t.Fatalf("[%s]: expecting %d executions, got: %d", n, tc.ExpectedExecutions, executions)
-		}
-	}
-}
-
 type ErrFatal struct{ error }
 
 func (ErrFatal) Fatal() bool {
@@ -167,17 +111,4 @@ func getMinExecutionTime(delayDuration time.Duration, executions int64, exponent
 		execTime = 0
 	}
 	return time.Duration(execTime)
-}
-
-func getBackoffMinTime(delay time.Duration, maxWaitTime time.Duration, executions int) time.Duration {
-	var duration time.Duration
-	for i := 1; i < executions; i++ {
-		duration += delay
-		delay *= 2
-		if delay > maxWaitTime {
-			delay = maxWaitTime
-		}
-	}
-
-	return duration
 }

